Oceanwater 626 ros actions for antenna

anjan1984 opened the pull request:
## Linked Issues:
| EPIC ⚡ | [OCEANWATER-XXX](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-XXX) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-626](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-626_ROS_Actions_for_antenna ) |
| Github :octocat:  | # |


## Summary of Changes
* Add ros action support for antenna movement.

## Test
* roslaunch ow europa_terminator_workspace.launch 
* In a new terminal 
* cd src/ow_simulator/ow_lander/scripts
* python antenna_action_client.py 

Modify antenna_action_client.py file for different pan and tilt values.

Samahu commented on ow_lander/scripts/antenna_action_server.py:
```diff
@@ -0,0 +1,105 @@
+import rospy
+import actionlib
+import ow_lander.msg
+# import sys
+# import copy
+# import moveit_commander
+# import moveit_msgs.msg
+# import geometry_msgs.msg
+#from math import pi
+#from std_msgs.msg import String 
```
remove unused imports

anjan1984 replied:
done!

Samahu commented on ow_lander/scripts/antenna_action_server.py:
```diff
@@ -0,0 +1,105 @@
+from sensor_msgs.msg import JointState
+from gazebo_msgs.msg import LinkStates
+from moveit_commander.conversions import pose_to_list
+# import math
+# import constants
+# import utils
+# import action_deliver_sample
+
+# from LanderInterface import MoveItInterface
+# from LanderInterface import LinkStateSubscriber
+# from trajectory_async_execution import TrajectoryAsyncExecuter
+# from moveit_msgs.msg import RobotTrajectory
```
remove unused imports

anjan1984 replied:
done!

Samahu reviewed (changes requested):
- [x] Remove unused imports
- [x] Consider increasing the tolerance value since I have seen cases where the movement doesn't converge!
- [x] It might be necessary to check if the angles wrap over 180 or 360 degrees (+ or -) when getting the difference value between angles.
- [ ] Recommended to rename the action to **AntennaPanTiltAction** and all dependent filenames to avoid confusion in the future.

anjan1984 commented on ow_lander/scripts/antenna_action_server.py:
```diff
@@ -0,0 +1,105 @@
+            self.tilt_pub.publish(goal.tilt) 
+            self._update_feedback()
+
+            if (abs(goal.pan - self._pan_value)< 0.01 and abs(goal.tilt - self._tilt_value)< 0.01):
```
done! made it 0.05

anjan1984 commented on ow_lander/scripts/antenna_action_server.py:
```diff
@@ -0,0 +1,105 @@
+        self._result = ow_lander.msg.AntennaResult()
+        self.tilt_pub = rospy.Publisher('/ant_tilt_position_controller/command', Float64, queue_size=10)
+        self.pan_pub = rospy.Publisher('/ant_pan_position_controller/command', Float64, queue_size=10)
+        self.subscriber = rospy.Subscriber("/joint_states", JointState, self._handle_joint_states)
```
done!

anjan1984 commented:
> * [ ]  Remove unused imports
> * [ ]  Consider increasing the tolerance value since I have seen cases where the movement doesn't converge!
> * [ ]  It might be necessary to check if the angles wrap over 180 or 360 degrees (+ or -) when getting the difference value between angles.
> * [ ]  Recommended to rename the action to **AntennaPanTiltAction** and all dependent filenames to avoid confusion in the future.

addressed all the suggestions above

kmdalal commented:
Got this message for the first time ever:
```
ServiceCaller.on_refresh_services_button_clicked(): could not get class of service /rviz_rockaway_ndc_nasa_gov_28555_3416582078087782726/save_config:
Unable to load type [rviz/SendFilePath].
Have you typed 'make' in [rviz]?
ServiceCaller.on_refresh_services_button_clicked(): could not get class of service /rviz_rockaway_ndc_nasa_gov_28555_3416582078087782726/load_config:
Unable to load type [rviz/SendFilePath].
Have you typed 'make' in [rviz]?
```

kmdalal reviewed (approved):
Aside from the new warning I reported, which I suspect is unrelated, this feature worked for me -- I tried several pan/tilt angles.  For regression I verified Plexil TestAntennaMovement, which uses topics.  Nice work!

Samahu commented on ow_lander/CMakeLists.txt:
```diff
@@ -35,6 +35,7 @@ add_action_files(
   FILES DigCircular.action
   FILES Grind.action
   FILES GuardedMove.action
+  FILES Antenna.action
```
I meant all references.  
```suggestion
  FILES AntennaPanTilt.action
```

kmdalal replied:
Good catch @Samahu.  I'll agree.  While the action names technically aren't all verbs (they are nouns and/or verbs), they do specifically mention the action and not just the component.

anjan1984 replied:
Done!

Samahu reviewed (changes requested):
When I mentioned to rename `Antenna.action` to `AntennaPanTilt.action` I mean all references not just the node name. There is two reasons I am requesting this:
1. To be consistent with the other action names: {`Stow`, `DigCircular`, `Grind`, `GuardedMove`, ...}. All of these actions are pronounced as verbs the only odd one is the `Antenna.action`.
2. Naming the action Antenna would be confusing if we later wanted to add an action that handles communication via the Antenna. To avoid confusion we should have: `AntennaPanTIlt.action` and maybe later on `AntennaSendReceive.action`

Samahu reviewed (approved):
I would have wished you renamed `antenna_action_client.py` to `antenna_pan_tilt_action_client.py` and `antenna_action_server.py` to `antenna_pan_tilt_action_server.py` but it is not a big deal.

kmdalal commented:
I agree with @Samahu 's script name suggestions, though this could be done when the next kind of action is added -- or now if you'd like.

anjan1984 commented:
> I would have wished you renamed `antenna_action_client.py` to `antenna_pan_tilt_action_client.py` and `antenna_action_server.py` to `antenna_pan_tilt_action_server.py` but it is not a big deal.

Done!
